pkg/crinit/aggregated/aggregated.go

```diff
@@ -0,0 +1,399 @@
+			}
+			hostClientset, err := client.NewForConfig(hostConfig)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+			apiServiceClientset, err := apiregclient.NewForConfig(hostConfig)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+			err = Run(opts, cmdOut, hostClientset, apiServiceClientset, pathOptions)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	opts.BindCommon(flags, defaultServerImage, defaultEtcdImage)
+	opts.Bind(flags)
+
+	return cmd
+}
+
+// validateOptions ensures that options are valid.
+func validateOptions(opts *aggregatedClusterRegistryOptions) error {
+	opts.ApiServerServiceType = v1.ServiceType(opts.apiServerServiceTypeString)
+	return opts.ValidateCommonOptions()
+}
+
+// Run initializes a cluster registry.
+func Run(opts *aggregatedClusterRegistryOptions, cmdOut io.Writer,
+	hostClientset client.Interface, apiSvcClientset apiregclient.Interface, pathOptions *clientcmd.PathOptions) error {
+
+	err := opts.CreateNamespace(cmdOut, hostClientset)
+	if err != nil {
+		return err
+	}
+
+	svc, ips, hostnames, err := opts.CreateService(cmdOut, hostClientset)
+	if err != nil {
+		return err
+	}
+
+	credentials, err := opts.GenerateCredentials(cmdOut, svc.Name, ips, hostnames,
+		false, false)
+	if err != nil {
+		return err
+	}
+
+	err = opts.CreateAPIServerCredentialsSecret(hostClientset, credentials)
+	if err != nil {
+		return err
+	}
+
+	var pvc *v1.PersistentVolumeClaim
+	pvc, err = opts.CreatePVC(cmdOut, hostClientset, svc.Name)
+	if err != nil {
+		return err
+	}
+
+	err = opts.CreateAPIServer(cmdOut, hostClientset, false, false, ips, pvc)
+	if err != nil {
+		return err
+	}
+
+	_, err = createAPIService(cmdOut, apiSvcClientset, opts,
+		util.GetCAKeyPair(credentials).Cert)
+	if err != nil {
+		return err
+	}
+
+	err = createRBACObjects(cmdOut, hostClientset, opts)
+	if err != nil {
+		return err
+	}
+
+	err = opts.UpdateKubeconfig(cmdOut, pathOptions, svc, ips, hostnames,
+		credentials)
+	if err != nil {
+		return err
+	}
+
+	return opts.WaitForAPIServer(cmdOut, hostClientset, pathOptions, ips,
+		hostnames, svc)
+}
+
+// createRbacObjects handles the creation of all the RBAC objects necessary
+// to deploy the cluster registry in aggregated mode
+func createRBACObjects(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) error {
+
+	fmt.Fprintf(cmdOut, "Creating RBAC objects...")
+
+	// Create a Kubernetes service account in your namespace
+	glog.V(4).Infof(
+		"Creating service account %v for cluster registry apiserver in the host cluster",
+		serviceAccount)
+
+	sa, err := createServiceAccount(cmdOut, clientset, opts)
+
+	if err != nil {
+		glog.V(4).Infof("Failed to create service account %v: %v", sa, err)
+		return err
+	}
+
+	glog.V(4).Info("Successfully created service account")
+
+	// Create a Kubernetes cluster role for the operations you want to allow
+	// on your resources
```

Comment:
nit: This comment seems pretty generic. Can you make it more specific to what you're doing here for the cluster registry?